flax/linen/recurrent.py

```diff
@@ -142,6 +142,146 @@ def initialize_carry(rng, batch_dims, size, init_fn=zeros):
     return init_fn(key1, mem_shape), init_fn(key2, mem_shape)
 
 
+class DummyDense(Module):
+  """Dummy module for creating parameters matching `flax.nn.Dense`."""
+
+  features: int
+  use_bias: bool = True
+  dtype: Dtype = jnp.float32
+  precision: Any = None
+  kernel_init: Callable[[PRNGKey, Shape, Dtype], Array] = default_kernel_init
+  bias_init: Callable[[PRNGKey, Shape, Dtype], Array] = zeros
+
+  @compact
```

Comment:
If you just need to define parameters, it's cleaner to use `setup`, e.g

```py
def setup(self):
  self.kernel = self.param('kernel', ...)
  self.bias = self.param('bias', ...)
```

(IIUC that you don't look at the return values from `__call__` at all)

Reply:
Discussed offline: will keep them in `__call__` because need input shape.